subsys/net/ip/net_context.c

```diff
@@ -258,7 +258,7 @@ static int tcp_backlog_ack(struct net_pkt *pkt, struct net_context *context)
 	context->tcp->send_seq = tcp_backlog[r].send_seq;
 	context->tcp->send_ack = tcp_backlog[r].send_ack;
 
-	if (k_delayed_work_cancel(&tcp_backlog[r].ack_timer) < 0) {
+	if (k_delayed_work_cancel(&tcp_backlog[r].ack_timer) == -EINPROGRESS) {
```

Comment:
Perhaps we should got with https://github.com/zephyrproject-rtos/zephyr/pull/1141 since k_delayed_work_cancel no longer return -EINPROGRESS, and -EAGAIN is fine since it just mean the work is idle.